integrations/pgvector/tests/test_document_store.py

```diff
@@ -215,3 +272,44 @@ def get_table_oid(document_store, schema_name, table_name):
     # Clean up: drop the schema after the test
     with psycopg.connect(connection_string, autocommit=True) as conn:
         conn.execute(f"DROP SCHEMA IF EXISTS {schema_name} CASCADE")
+
+
+@pytest.mark.integration
+@requires_pgvector_0_7
+def test_halfvec_hnsw_high_dimension():
+    """
+    Test that PgvectorDocumentStore works with vector_type='halfvec',
+    search_strategy='hnsw', and dimensions > 2000.
+    """
+    high_dimension = 3072
+    test_schema_name = "test_halfvec_schema"
+    test_table_name = "test_halfvec_table"
+    connection_string = "postgresql://postgres:postgres@localhost:5432/postgres"
+
+    try:
+        with psycopg.connect(connection_string, autocommit=True) as conn:
+            conn.execute(f"DROP SCHEMA IF EXISTS {test_schema_name} CASCADE")
+            conn.execute(f"CREATE SCHEMA {test_schema_name}")
+
+        document_store = PgvectorDocumentStore(
+            connection_string=Secret.from_token(connection_string),
+            schema_name=test_schema_name,
+            table_name=test_table_name,
+            embedding_dimension=high_dimension,
+            vector_type="halfvec",
+            search_strategy="hnsw",
+            vector_function="cosine_similarity",
+            recreate_table=True,
+            hnsw_recreate_index_if_exists=True,
+        )
+
+        # Ensure the database setup is complete
+        document_store._ensure_db_setup()
+
+        # Verify the column type
+        col_type = get_column_type(document_store, test_schema_name, test_table_name, "embedding")
+        assert col_type == "halfvec", f"Expected column type 'halfvec', but found '{col_type}'"
```

Comment:
Instead of doing this, I would try to
1. write a `Document` with >2000 dimension.
2. retrieve all Documents using `filter_documents`
3. verify that the original `Document` is present and the embedding has the expected dimension/